### PR TITLE
Router

### PR DESCRIPTION
**Pull request by ajandorek**

### CODE REVIEW

File: src/components/Forecast.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import ForecastDay from './ForecastDay';
+
+const Forecast = props => {
+  const { apiResponse, forecastData } = props;
+  if (apiResponse === false) return <div>Loading</div>;
+  return (
+    <div className="forecast">
+      {forecastData.list.map((day, i) => (
+        <ForecastDay key={i} weather={day} isCelcius={props.isCelcius} />
```
Comment (lucas-aragno) [inline src/components/Forecast.js]: can you add `isCelcius` to the destructing on top?

File: src/components/ForecastDay.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import moment from 'moment';
+import { renderWeatherImg, toCelcius } from '../utils/weather.js';
+
+const ForecastDay = props => (
+  <div className="forecastday">
+    <h2>{moment.unix(props.weather.dt).format('ddd')}</h2>
+    <h3>
+      {props.isCelcius ? `${toCelcius(props.weather.temp.day)}° C` : `${props.weather.temp.day}° F`}
```
Comment (lucas-aragno) [inline src/components/ForecastDay.js]: can you destructure props?

File: src/components/UnitToggle.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import ToggleSwitch from '@trendmicro/react-toggle-switch';
+
+const UnitToggle = props => (
+  <div>
+    F
+    <ToggleSwitch
+      size="sm"
+      checked={props.checkUnit}
+      onChange={() => {
```
Comment (lucas-aragno) [inline src/components/UnitToggle.js]: you can do `onChange={unitChange}` (if you destructure props)

File: src/utils/helpers.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export const titleCase = str =>
+  str
+    .split(' ')
+    .map(word => word.replace(word[0], word[0].toUpperCase()))
```
Comment (lucas-aragno) [inline src/utils/helpers.js]: what is this function doing?

**Review by lucas-aragno — CHANGES_REQUESTED** [review]

File: src/utils/helpers.js (hunk 4 above)
Comment (ajandorek) [reply src/utils/helpers.js]: The default response for current weather returns a string in lowercase so this makes it "title case"

File: src/components/Forecast.js (hunk 1 above)
Comment (ajandorek) [reply src/components/Forecast.js]: done

File: src/components/ForecastDay.js (hunk 2 above)
Comment (ajandorek) [reply src/components/ForecastDay.js]: done

File: src/components/UnitToggle.js (hunk 3 above)
Comment (ajandorek) [reply src/components/UnitToggle.js]: done

File: src/utils/helpers.js (hunk 4 above)
Comment (lucas-aragno) [reply src/utils/helpers.js]: can you use CSS for this?

Comment (ajandorek) [reply src/utils/helpers.js]: Ahh yes I see now

File: src/components/UnitToggle.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import ToggleSwitch from '@trendmicro/react-toggle-switch';
+
+const UnitToggle = props => {
+  const { unitChange, checkUnit } = props;
```
Comment (lucas-aragno) [inline src/components/UnitToggle.js]: can you destructure the props on the arguments

File: src/components/Weather.js [hunk 6]
```diff
@@ -1,55 +1,17 @@
...
-    super();
-    this.state = {
-      weatherData: null,
-      apiResponse: false,
-      temperature: null,
-      isCelcius: false,
-    };
-  }
+const Weather = props => {
+  const { apiResponse, weatherData, temperature } = props;
```
Comment (lucas-aragno) [inline src/components/Weather.js]: ditto

File: src/utils/location.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+const getCurrentPosition = (options = {}) =>
+  new Promise((resolve, reject) => {
+    navigator.geolocation.getCurrentPosition(resolve, reject, options);
+  });
+
+export const getLocation = async () => {
+  let position;
+  try {
```
Comment (lucas-aragno) [inline src/utils/location.js]: I think ashley wrote a comment about this block

**Review by lucas-aragno — CHANGES_REQUESTED** [review]

File: src/components/UnitToggle.js (hunk 5 above)
Comment (ajandorek) [reply src/components/UnitToggle.js]: I'm confused on what else needs destructuring here?

File: src/components/Weather.js (hunk 6 above)
Comment (ajandorek) [reply src/components/Weather.js]: done

File: src/components/App.js [hunk 8]
```diff
@@ -1,15 +1,118 @@
...
+        });
+        this.getCurrentWeather();
+        this.getCurrentForecast();
+      });
+    }
+  }
+
+  getCurrentWeather() {
+    const { latitude, longitude, location } = this.state;
+    const url = `http://api.openweathermap.org/data/2.5/weather?lat=${latitude}&lon=${longitude}&appid=${
```
Comment (lucas-aragno) [inline src/components/App.js]: I think you can maybe create a function `getServiceUrl(service)` that returns this url and where `service` is either `weather` or `forecast/daily`

File: src/components/App.js [hunk 9]
```diff
@@ -1,15 +1,118 @@
...
+      process.env.OPENWEATHER_APIKEY
+    }&units=imperial`;
+    if (location) {
+      getWeatherInformation(url, latitude, longitude).then(response => {
+        this.setState({
+          weatherData: response.data,
```
Comment (lucas-aragno) [inline src/components/App.js]: maybe destructure response?

File: src/components/App.js [hunk 10]
```diff
@@ -1,15 +1,118 @@
...
+    }
+  }
+
+  getCurrentForecast() {
+    const { latitude, longitude, location } = this.state;
+    const url = `http://api.openweathermap.org/data/2.5/forecast/daily?lat=${latitude}&lon=${longitude}&appid=${
+      process.env.OPENWEATHER_APIKEY
+    }&cnt=5&units=imperial`;
+    if (location) {
+      getWeatherInformation(url, latitude, longitude).then(response => {
```
Comment (lucas-aragno) [inline src/components/App.js]: ditto

File: src/components/App.js [hunk 11]
```diff
@@ -1,15 +1,118 @@
...
+        <BrowserRouter>
+          <div>
+            <UnitToggle unitChange={this.handleUnitChange} checkUnit={this.state.isCelcius} />
+            <Nav />
+            <Route
+              exact
+              path="/"
+              render={() => (
+                <Weather
+                  isCelcius={this.state.isCelcius}
```
Comment (lucas-aragno) [inline src/components/App.js:90]: destructure state

**Review by lucas-aragno — CHANGES_REQUESTED** [review]

File: src/components/App.js (hunk 8 above)
Comment (ajandorek) [reply src/components/App.js]: done

File: src/components/App.js (hunk 10 above)
Comment (ajandorek) [reply src/components/App.js]: done

File: src/components/App.js (hunk 9 above)
Comment (ajandorek) [reply src/components/App.js]: done

File: src/components/App.js (hunk 11 above)
Comment (ajandorek) [reply src/components/App.js:90]: done

File: src/components/ForecastDay.js [hunk 12]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import moment from 'moment';
```
Comment (lucas-aragno) [inline src/components/ForecastDay.js]: you can just `import {unix} from 'moment'`

Comment (ajandorek) [reply src/components/ForecastDay.js]: @lucas-aragno

**Pull request merged**